pkg/daemon/update.go

```diff
@@ -470,15 +470,31 @@ func (dn *Daemon) writeFiles(files []ignv2_2types.File) error {
 	return nil
 }
 
-// seperated out writeSSHKeys from updatePasswd even though this is the only update allowed to Passwd right now
+// Update a given PasswdUser's SSHKey
 func (dn *Daemon) writeSSHKeys(oldUsers []ignv2_2types.PasswdUser, newUsers []ignv2_2types.PasswdUser, userKeys []int) error {
 	var path string
 	for _, idx := range userKeys {
 		if !reflect.DeepEqual(oldUsers[idx].SSHAuthorizedKeys, newUsers[idx].SSHAuthorizedKeys) {
-			// ssh keys are in PasswdUser.HomeDir/.ssh
+			// ssh keys are in PasswdUser.HomeDir/.ssh/authorized_keys
 			path = filepath.Join(oldUsers[idx].HomeDir, ".ssh")
 			glog.V(2).Infof("Writing SSHKeys at %q:", path)
-			// write the SSHKeys
+			if err := dn.fileSystemClient.MkdirAll(filepath.Dir(path), os.FileMode(0600)); err != nil {
+				return fmt.Errorf("Failed to create directory %q: %v", filepath.Dir(path), err)
+			}
+			glog.V(2).Infof("Created directory: %s", path)
+			authkeypath := filepath.Join(path, "authorized_keys")
+
+			var concatSSHKeys string
+			for _, k := range newUsers[idx].SSHAuthorizedKeys {
+				concatSSHKeys = concatSSHKeys + string(k) + "/n"
```

Comment:
Why `/n` and not `\n`?

Reply:
DOH

Reply:
@rphillips Thank you for spotting that!

Reply:
@rphillips ah just remembered that I tried strings.Join before, but it won't work bc SSHAuthorizedKeys -> type []SSHAuthorizedKey -> type SSHAuthorizedKey string